### PR TITLE
MDX v2: unit tests

### PR DESCRIPTION
**Pull request by axe312ger**

### CODE REVIEW

File: packages/gatsby-plugin-mdx/src/__tests__/compile-mdx.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  panicOnBuild: jest.fn(e => {
+    throw e
+  }),
+} as unknown as NodePluginArgs["reporter"]
+
+describe(`compiles MDX`, () => {
+  it(`default`, async () => {
+    console.log(exampleMdxContent.toString())
+    try {
+      await compileMDX(
```
Comment (axe312ger) [inline packages/gatsby-plugin-mdx/src/__tests__/compile-mdx.ts:40]: I get `ModuleNotFoundError: Cannot find module 'estree-walker' from 'node_modules/estree-util-build-jsx/lib/index.js'`

Super weird, `estree-walker'` is there on my FS.

File: packages/gatsby-plugin-mdx/src/__tests__/remark-infer-toc-meta.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Some text with *formatting*.
+
+## Headline 2
+
+With some text beneath`
+
+describe(`remark: infer ToC meta`, () => {
+  it(`parses ToC and attaches it to our meta object`, async () => {
+    const processor = remark().use(remarkInferTocMeta, {
```
Comment (axe312ger) [inline packages/gatsby-plugin-mdx/src/__tests__/remark-infer-toc-meta.ts:17]: the `remarkInferTocMeta` is not accepted here as plugin. 

```
Argument of type 'Plugin<[InferTocMetaParams], Node, Node>' is not assignable to parameter of type 'Plugin<[{ toc: (node: Node, options?: Options | undefined) => Result; visit: (<Tree extends any, Check extends Test>(tree: Tree, test: Check, visitor: BuildVisitor<Tree, Check>, reverse?: boolean | undefined) => void) & (<Tree_1 extends any>(tree: Tree_1, visitor: BuildVisitor<...>, reverse?: boolean | undefined) =>...'.
  The 'this' types of each signature are incompatible.
    Property 'attachers' is missing in type 'Processor<RemarkOptions>' but required in type 'Processor<void, Node, void, void>'.ts(2345)
index.d.ts(238, 3): 'attachers' is declared here.
```

File: packages/gatsby-plugin-mdx/src/__tests__/get-source-plugins-as-remark-plugins.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { getSourcePluginsAsRemarkPlugins } from "../get-source-plugins-as-remark-plugins"
+import { mockGatsbyApi } from "../__fixtures__/test-utils"
+
+const { getNode, getNodesByType, reporter, cache, pathPrefix } = mockGatsbyApi()
+
+describe(`transform gatsby-remark plugins into regular plugins`, () => {
+  it(`skips on non-functionals`, async () => {
+    await expect(
+      getSourcePluginsAsRemarkPlugins({
+        gatsbyRemarkPlugins: [],
```
Comment (axe312ger) [inline packages/gatsby-plugin-mdx/src/__tests__/get-source-plugins-as-remark-plugins.ts:10]: @todo

File: packages/gatsby-plugin-mdx/src/__tests__/remark-mdx-html-plugin.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+`
+
+describe(`remark: infer ToC meta`, () => {
+  it(`parses ToC and attaches it to our meta object`, async () => {
+    const processor = unified()
+      .use(remarkParse)
+      .use(remarkMdx)
+      .use(remarkMdxHtmlPlugin)
+      .use(remarkStringify)
+    const res = await processor.process(source)
```
Comment (axe312ger) [inline packages/gatsby-plugin-mdx/src/__tests__/remark-mdx-html-plugin.ts:20]: This runs forever and then after 5 seconds:

`Cannot handle unknown node `mdxJsxFlowElement``

**Review by LekoArts — APPROVED** [review]

**Pull request merged**